lib/db.js

```diff
@@ -132,19 +132,18 @@ const legalOptionNames = [
  * @property {object} topology Access the topology object (single server, replicaset or mongos).
  * @returns {Db} a Db instance.
  */
+
 function Db(databaseName, topology, options) {
   options = options || {};
   if (!(this instanceof Db)) return new Db(databaseName, topology, options);
 
-  // Get the promiseLibrary
-  const promiseLibrary = options.promiseLibrary || Promise;
+  if (options.promiseLibrary) {
+    emitDeprecationWarning('MongoClient option [promiseLibrary] is deprecated');
```

Comment:
You're emitting this deprecation warning in `Db` not `MongoClient`, also you are only emitting this in `Db`, but we used to accept it as an option in a few other types?

Reply:
Added it to:
-  lib/collection.js 
-  lib/cursor/cursor.js 
-  lib/db.js 
-  lib/sdam/topology.js